DOCS/man/vo.rst

```diff
@@ -363,6 +363,76 @@ Available video output drivers are:
     ``--vo-tct-256=<yes|no>`` (default: no)
         Use 256 colors - for terminals which don't support true color.
 
+``sixel``
+    Sixel graphics video output driver based on libsixel that works on a
+    console that has sixel graphics enabled such as ``xterm`` or ``mlterm``.
+    Additionally some terminals have limitation on the dimensions, so may
+    not display images bigger than 1000x1000 for example. Make sure that
+    ``img2sixel`` can display images of the corresponding resolution.
+
+    Since mpv 0.30.0, you may need to use ``--profile=sw-fast`` to get decent
+    performance.
+
+    Note: the Sixel image output is not synchronized with other terminal output
+    from mpv, which can lead to broken images. The option ``--really-quiet``
+    can help with that, and is recommended.
+
+    ``--vo-sixel-diffusion=<algo>``
+        Selects the diffusion algorithm for dithering used by libsixel.
+        Can be one of the below list as per libsixel's documentation.
+
+        auto
+            Choose diffuse type automatically
+        none
+            Don't diffuse
+        atkinson
+            Diffuse with Bill Atkinson's method. (Default)
+        fs
+            Diffuse with Floyd-Steinberg method
+        jajuni
+            Diffuse with Jarvis, Judice & Ninke method
+        stucki
+            Diffuse with Stucki's method
+        burkes
+            Diffuse with Burkes' method
+        arithmetic
+            Positionally stable arithmetic dither
+        xor
+            Positionally stable arithmetic xor based dither
+
+    ``--vo-sixel-width=<width>``  ``--vo-sixel-height=<height>``
```

Comment:
How does the size in pixels relate to the size in character cells? can it default to fill the terminal size until the left/right edges or the top/bottom (sans 1 line) touch the edges, while the other edges are centered (similar to tct)?

Reply:
If we use the ioctl `TIOCGWINSZ` then we should be able to get the dimensions of the terminal window in pixels as well as the character width and height as well. So should be able to calculate the optimal size of the terminal window `(height_of_terminal * (num_rows -1 )/num_rows)` While for the width we can select the terminal full width generally.

Reply:
You don't need to use it directly. Mpv already has `terminal_get_size`, and you only need to include `#include "osdep/terminal.h"` - like `vo_tct.c` does.

If that's the only only missing thing to make the video fit automatically, then maybe we should change the width/height value be in character cells units rather than pixels?

Do you need to figure out how many pixels are in each character? Again, who figures the relation between character cells to pixels, and how? what happens if the terminal changes the font size (and keeping the size in cells) while a video is playing?

Reply:
`terminal_get_size` just returns the cell counts like `80x25`. That is not enough info to map to pixels. Like the terminal on a 1920x1080 monitor might have different pixel width, height compared to a 4k monitor. It also depends on the font size of the terminal set. 
 That is why the ioctl call is required to get the pixels of terminal window. Plus on resizing the terminal window, you could make the bigger image fit if you started with a smaller terminal initially. 
Also then aspect ratio might have to be preserved along with the terminal pixel dimensions. So it is a bit tricky part.
Therefore currently it is fully user specified which I agree is not good user experience.